api/models/Notification.js

```diff
@@ -134,46 +134,50 @@ module.exports = bookshelf.Model.extend({
   sendEventInvitationPush: function () {
     const post = this.post()
     const actor = this.actor()
+    const locale = this.reader().get('settings')?.locale || this.actor().get('settings')?.locale || 'en'
     const groupIds = Activity.groupIds(this.relations.activity)
     if (isEmpty(groupIds)) throw new Error('no group ids in activity')
     return Group.find(groupIds[0])
       .then(group => {
         var path = url.parse(Frontend.Route.post(post, group)).path
-        var alertText = PushNotification.textForEventInvitation(post, actor)
+        var alertText = PushNotification.textForEventInvitation(post, actor, locale)
         return this.reader().sendPushNotification(alertText, path)
       })
   },
 
   sendPushAnnouncement: function (version) {
     var post = this.post()
     var groupIds = Activity.groupIds(this.relations.activity)
+    const locale = this.reader().get('settings')?.locale || this.actor().get('settings')?.locale || 'en'
     if (isEmpty(groupIds)) throw new Error('no group ids in activity')
     return Group.find(groupIds[0])
       .then(group => {
         var path = url.parse(Frontend.Route.post(post, group)).path
-        var alertText = PushNotification.textForAnnouncement(post)
+        var alertText = PushNotification.textForAnnouncement(post, locale)
         return this.reader().sendPushNotification(alertText, path)
       })
   },
 
   sendPostPush: function (version) {
     var post = this.post()
     var groupIds = Activity.groupIds(this.relations.activity)
+    const locale = this.reader().get('settings')?.locale || this.actor().get('settings')?.locale || 'en'
```

Comment:
seems like this should be turned into a helper function

Reply:
ok